ui/libproject/devstart.rb

```diff
@@ -0,0 +1,94 @@
+require "optparse"
+require_relative "utils/common"
+
+def install_dependencies()
+  common = Common.new
+  common.docker.requires_docker
+
+  parent = Dir.chdir("..") { Dir.pwd }
+
+  common.run_inline %W{docker run --rm -w /w/ui -v #{parent}:/w:cached node npm install}
+end
+
+def swagger_regen()
+  common = Common.new
```

Comment:
Doing this outside of the docker container means you have to manually remove the `generated` dir if you're making changes to the swagger file? This feels a bit awkward, though I'm not sure I have a better suggestion. One idea (which I did in circle) is to mount the repo root into the docker container, and then the swagger file is visible from within the ui container.